### PR TITLE
Fix focus state on licence-finder journey

### PR DESCRIPTION
**Pull request by maxgds**
https://trello.com/c/uJOqVUmd/83-update-license-finder-to-govukpublishingcomponents-v21-with-compatibility-mode-on

This PR tackles improving the focus state throughout the licence finder journey starting at 
https://www.gov.uk/licence-finder/sectors

This involves:
Updating the components gem and removing legacy mixins
Adding the govuk-link class to various links
Replacing the buttons with component versions (and adjusting the js to accommodate this)
Tweaking the layout where needed to support the focus state

*Out of scope for this work*
Improving the strange design choices - odd layout including spurious padding has been mostly left as it was
Updating the CSS to avoid element selectors
Replacing form elements with components

### Before/after screenshots
Screenshot 2019-11-27 at 10 22 30
Screenshot 2019-11-27 at 10 22 12

-----

Screenshot 2019-11-27 at 10 23 30
Screenshot 2019-11-27 at 10 23 39
-----
Screenshot 2019-11-27 at 10 24 30
Screenshot 2019-11-27 at 10 24 38
-----
Screenshot 2019-11-27 at 11 06 43
Screenshot 2019-11-27 at 11 06 53
-----
Screenshot 2019-11-27 at 11 08 08
Screenshot 2019-11-27 at 11 08 16
-----
Screenshot 2019-11-27 at 11 10 12
Screenshot 2019-11-27 at 11 10 22
-----
Screenshot 2019-11-27 at 11 11 06
Screenshot 2019-11-27 at 11 11 15
-----

### CODE REVIEW

File: app/views/licence_finder/activities.html.erb [hunk 1]
```diff
@@ -30,11 +30,14 @@
         <%= basket_link activity %>
       <% end %>
     </ul>
-    <% unless @picked_activities.empty? %>
-      <div class="button-container">
-        <%= link_to 'Next step', business_location_path(sectors: params[:sectors], activities: params[:activities]), class: 'button medium', id: "next-step"  %>
-      </div>
-    <% end %>
+
+    <div class="button-container js-hidden"><!-- FIXME - what about non-js? -->
```
Comment (maxgds) [inline app/views/licence_finder/activities.html.erb]: oops, need to kill that comment

File: app/views/licence_finder/sectors.html.erb [hunk 2]
```diff
@@ -6,7 +6,9 @@
       <%= hidden_field_tag "sectors", params[:sectors], id: "hidden-sectors" %>
       <label class="visuallyhidden" for="search-sectors">Search for areas</label>
       <%= text_field_tag "q", params[:q], id: "search-sectors" %>
```
[the remark this answers is not in the document]
Comment (maxgds) [reply app/views/licence_finder/sectors.html.erb]: Yeah, I thought the remit was to make links work and address other components later, though at this stage? Do you think I need to do both here?

File: spec/support/custom_matchers.rb [hunk 3]
```diff
@@ -57,7 +57,9 @@ def check_add_remove_link(type, label)
     element = find(:xpath, ".//li[span/text() = '#{label}']")
     expect(element).not_to be_nil
     expect(element).to have_xpath("a[text() = '#{type}']")
-    expect(element).to have_xpath("a[@class = '#{type_class}']")
+
+    expect(element).to have_selector(:css, "a.govuk-link.#{type_class}")
+    #expect(element).to have_xpath("a[@class = 'govuk-link #{type_class}']")
```
Comment (maxgds) [inline spec/support/custom_matchers.rb]: ugh, and this one needs removing

File: app/views/licence_finder/sectors.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (injms) [reply app/views/licence_finder/sectors.html.erb]: I think that it's worth making the new focus states work on all elements that need it, including input elements. Whether that's by using the input component from the gem or by adding a `govuk-` class to the input doesn't matter. 🙂

**Comment by maxgds** [comment]
@injms I've addressed all issues, including replacing the form input and the form dropdown with components, please take another look when you have time.

**Pull request merged**